tablo/geom_utils.py

```diff
@@ -437,4 +437,196 @@ def extract_significant_digits(number):
+    # ESRI geometry JSON to GeoJSON
+    # TODO look for optimization possibilities
+
+    @staticmethod
+    def rings_to_wkt(rings):
+        outer_rings = []
+        holes = []
+
+        for r in rings:
+            ring = Terraformer.close_ring(r)
+            if len(ring) < 4:
+                continue
+
+            if Terraformer.ring_is_clockwise(ring):
+                # a clockwise ring means its an outer ring
+                polygon = [ring]
+                outer_rings.append(polygon)
+            else:
+                # otherwise it's a hole!
+                holes.append(ring)
+
+        uncontained_holes = []
+
+        while holes:
+            hole = holes.pop()
+
+            # loop over all outer rings and see if they contain our hole.
+            contained = False
+            for r in reversed(outer_rings):     # TODO why esri loops over a reversed rings?
+                outer_ring = r[0]
+                if Terraformer.coordinates_contain_coordinates(outer_ring, hole):
+                    r.append(hole)
+                    contained = True
+                    break
+
+            # sometimes a ring may not be contained in any outer ring (https://github.com/Esri/esri-leaflet/issues/320)
+            if not contained:
+                uncontained_holes.append(hole)
+
+        # check and see if uncontained holes intersect with any outer rings
+        while uncontained_holes:
+            hole = uncontained_holes.pop()
+
+            intersects = False
+            for r in reversed(outer_rings):     # TODO why esri loops over a reversed rings?
+                outer_ring = r[0]
+                if Terraformer.arrays_intersect_arrays(outer_ring, hole):
+                    outer_rings[r].push(hole)
+                    intersects = True
+                    break
+
+            # hole does not intersect ANY outer ring at this point; make it an outer ring.
+            if not intersects:
+                hole.reverse()
+                outer_rings.append([hole])
+
+        return {
+            'type': 'MultiPolygon',
+            'coordinates': outer_rings
+        }
+
+    @staticmethod
+    def close_ring(coordinates):
+        # makes sure the ring is closed
+        if coordinates[0] != coordinates[-1]:
+            coordinates.push(coordinates[0])
+
+        return coordinates
+
+    @staticmethod
+    def ring_is_clockwise(ring_to_test):
+        # a positive sum of areas under each side of a polygon indicates a clockwise polygon
+        # http://stackoverflow.com/questions/1165647/how-to-determine-if-a-list-of-polygon-points-are-in-clockwise-order
+        total = 0
+        r_length = len(ring_to_test)
+        for idx, pt1 in enumerate(ring_to_test):
+            if idx + 1 != r_length:
+                pt2 = ring_to_test[idx + 1]
+                total += (pt2[0] - pt1[0]) * (pt2[1] + pt1[1])
+        return total >= 0
```

Comment:
Would it be possible to break from the loop as soon as `total` is positive, or does the loop need to iterate over all of them to be sure?

Also, what does `pt1` and `pt2` stand for? Can there be a comment, or an improvement to the variable names to make it clearer?